sdks/java/core/src/main/java/org/apache/beam/sdk/util/BigQueryServices.java

```diff
@@ -67,4 +83,51 @@ void startQueryJob(String jobId, JobConfigurationQuery query, boolean dryRun)
     Job pollJob(String projectId, String jobId, int maxAttempts)
         throws InterruptedException, IOException;
   }
+
+  /**
+   * An interface for the Cloud BigQuery table service.
+   */
+  public interface TableService {
+    /**
+     * Gets the specified table resource by table ID.
+     */
+    Table getTable(String projectId, String datasetId, String tableId)
+        throws InterruptedException, IOException;
+
+    Dataset createDataset(String projectId, String datasetId, String location, String description)
```

Comment:
Javadoc. Also, can you please clearly state what invariants and failure behavior these functions have (possibly, in the impl classes instead).